File: frontend/src/api/mood.ts
import axios from 'axios'

const api = axios.create({ baseURL: '/api', timeout: 60000 })

export interface MoodDay {
  date: string
  mood_label: string
  mood_score?: number | null
  count: number
  source_count: Record<string, number>
}

export interface MoodSummary {
  days: MoodDay[]
  total_count: number
  mood_count: Record<string, number>
}

export const getMoodCalendar = (sid: string, year: number, month: number) =>
  api.get<MoodSummary>(`/mood/calendar?session_id=${sid}&year=${year}&month=${month}`)

File: backend/app/schemas/mood.py
from datetime import date

from pydantic import BaseModel


class MoodDay(BaseModel):
    date: date
    mood_label: str
    mood_score: float | None = None
    count: int
    source_count: dict[str, int]


class MoodSummary(BaseModel):
    days: list[MoodDay]
    total_count: int
    mood_count: dict[str, int]
